### PR TITLE
feature/HT20-246 E2E & Gateway & Usecase tests addition

### PR DESCRIPTION
**Pull request by venkatrangala**
https://hackney.atlassian.net/browse/HT20-246

Added Tests for the Post End point

### CODE REVIEW

File: AssetInformationApi.Tests/V1/UseCase/AddAssetUseCaseTests.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        public async Task AddAssetUsecaseShouldReturnOkResponse()
+        {
+            var asset = _fixture.Create<Asset>();
+            asset.Id = Guid.NewGuid();
+
+            var request = asset.ToDatabase();
+            _mockGateway.Setup(x => x.AddAsset(request)).ReturnsAsync(asset);
+
+            var response = await _classUnderTest.PostAsync(asset).ConfigureAwait(false);
+            //response.Should().BeEquivalentTo(asset.ToResponse());
```
Comment (duncanheron) [inline AssetInformationApi.Tests/V1/UseCase/AddAssetUseCaseTests.cs:37]: assertion commented out

Comment (venkatrangala) [reply AssetInformationApi.Tests/V1/UseCase/AddAssetUseCaseTests.cs:37]: yes spoke to Humairaa and will sort out on Monday.

**Review by rcmsoftwaredevelopment — APPROVED** [review]

File: AssetInformationApi.Tests/V1/E2ETests/Steps/AddNewAssetSteps.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        public async Task ThenTheAssetDetailsAreReturned(Asset request)
+        {
+            _lastResponse.StatusCode.Should().Be(HttpStatusCode.Created);
+            var responseContent = await _lastResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+            var apiProcess = JsonConvert.DeserializeObject<Asset>(responseContent);
+
+            apiProcess.Id.Should().NotBeEmpty();
+            var dbRecord = await _dbFixture.DynamoDbContext.LoadAsync<AssetDb>(apiProcess.Id).ConfigureAwait(false);
+
+            dbRecord.Id.Should().Be(request.Id);
```
Comment (humulla) [inline AssetInformationApi.Tests/V1/E2ETests/Steps/AddNewAssetSteps.cs:39]: Can you please add an assertion for the rest of the objects to ensure they are the same as the request

File: AssetInformationApi.Tests/V1/E2ETests/Stories/AddNewAssetTests.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            if (disposing && !_disposed)
+            {
+                if (null != _assetsFixture)
+                    _assetsFixture.Dispose();
+
+                _disposed = true;
+            }
+        }
+
+        [Fact]
```
Comment (humulla) [inline AssetInformationApi.Tests/V1/E2ETests/Stories/AddNewAssetTests.cs:47]: Please add more tests for Bad Request and NotFound scenarios

File: AssetInformationApi.Tests/V1/UseCase/AddAssetUseCaseTests.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        private readonly NewAssetUseCase _classUnderTest;
+        private readonly Fixture _fixture = new Fixture();
+
+        public NewAssetUseCaseTests()
+        {
+            _mockGateway = new Mock<IAssetGateway>();
+            _classUnderTest = new NewAssetUseCase(_mockGateway.Object);
+        }
+
+        [Fact]
```
Comment (humulla) [inline AssetInformationApi.Tests/V1/UseCase/AddAssetUseCaseTests.cs:27]: Please add another tests for when there is an application exception

**Review by humulla — COMMENTED** [review]
If there are any entities that are mandatory to be provided in the request object from the client please add FluentValidation and relative tests for them as well

File: AssetInformationApi.Tests/V1/E2ETests/Stories/AddNewAssetTests.cs (hunk 3 above)
Comment (venkatrangala) [reply AssetInformationApi.Tests/V1/E2ETests/Stories/AddNewAssetTests.cs:47]: Added for BadRequest, notfound is not valid for Post end points

File: AssetInformationApi/V1/Boundary/Request/AddAssetRequest.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+using Hackney.Shared.Asset.Domain;
+
+namespace AssetInformationApi.V1.Boundary.Request
+{
+    public class AddAssetRequest : Asset
```
Comment (humulla) [inline AssetInformationApi/V1/Boundary/Request/AddAssetRequest.cs:5]: Why do we have an empty request object? 
The request object should be in the Asset Shared Nuget Package

**Pull request merged**